src/argilla/client/feedback/dataset/remote/dataset.py

```diff
@@ -480,6 +481,46 @@ def add_records(
         """
         self._records.add(records=records, show_progress=show_progress)
 
+    @allowed_for_roles(roles=[UserRole.owner, UserRole.admin])
+    def search_by(
+        self,
+        vector_name: str,
+        value: Optional[List[float]] = None,
+        record: Optional[RemoteFeedbackRecord] = None,
```

Comment:
@alvarobartt is this signature ok? When we expect a record from the outside world it should be a `RemoteFeedbackRecord`?

Reply:
No, users will never instantiate a `RemoteFeedbackRecord` but a `FeedbackRecord`, if we are planning to allow both then we should use the `Union`

Reply:
Ok, I will change it to use `FeedbackRecord`.

Reply:
Since we need the `record_id` to find similar records for a given record, we need records already stored in Argilla, so `RemoteFeedbackRecord` objects. 

This method cannot work with records without an ID. The workflow to work with records in this method could be:
```python
argilla_ds = FeedbackDataset.from_argilla("dataset")

reference_record = argilla_ds[0]

simlar_records = argilla_ds.search_by(vector_name="my_vector", record=reference_record)

```